cmd/bundle/main.go

```diff
@@ -345,6 +365,134 @@ func dockerHubRepoDelete(token string, ref name.Reference) error {
+		return fmt.Errorf("failed to delete image %q: %s (HTTP status code %d)",
+			ref.String(),
+			string(respData),
+			resp.StatusCode,
+		)
+	}
+}
+
+func icrLogin(apikey string) (string, string, error) {
+	data := fmt.Sprintf("grant_type=%s&apikey=%s",
+		url.QueryEscape("urn:ibm:params:oauth:grant-type:apikey"),
+		apikey,
+	)
+
+	req, err := http.NewRequest("POST", "https://iam.cloud.ibm.com/identity/token", strings.NewReader(data))
+	if err != nil {
+		return "", "", err
+	}
+
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Accept", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", "", err
+	}
+
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", "", err
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		type ibmCloudIdentityToken struct {
+			AccessToken  string `json:"access_token"`
+			RefreshToken string `json:"refresh_token"`
+			TokenType    string `json:"token_type"`
+			Scope        string `json:"scope"`
+			ExpiresIn    int64  `json:"expires_in"`
+			Expiration   int64  `json:"expiration"`
+		}
+
+		var identityToken ibmCloudIdentityToken
+		if err := json.Unmarshal(body, &identityToken); err != nil {
+			return "", "", err
+		}
+
+		var token = fmt.Sprintf("%s %s", identityToken.TokenType, identityToken.AccessToken)
+
+		var accountID string
+		_, _ = jwt.Parse(identityToken.AccessToken, func(t *jwt.Token) (interface{}, error) {
+			switch obj := t.Claims.(type) {
+			case jwt.MapClaims:
+				if account, ok := obj["account"]; ok {
+					switch accountMap := account.(type) {
+					case map[string]interface{}:
+						switch tmp := accountMap["bss"].(type) {
+						case string:
+							accountID = tmp
+						}
+					}
+				}
+			}
+
+			return nil, nil
+		})
+
+		if accountID == "" {
+			return "", "", fmt.Errorf("failed to obtain account ID from identity token")
+		}
+
+		return token, accountID, nil
+
+	default:
+		var responseMsg map[string]interface{}
+		if err := json.Unmarshal(body, &responseMsg); err != nil {
+			return "", "", err
+		}
+
+		errorCode, errorCodeFound := responseMsg["errorCode"]
+		errorMessage, errorMessageFound := responseMsg["errorMessage"]
+		if errorCodeFound && errorMessageFound {
+			return "", "", fmt.Errorf("failed to obtain identity token from IAM: %v (%v)", errorMessage, errorCode)
+		}
+
+		return "", "", fmt.Errorf("failed to obtain identity token from IAM: %s", string(body))
+	}
+}
+
+func icrDelete(token string, accountID string, ref name.Reference) error {
+	deleteUrl := fmt.Sprintf("https://%s/api/v1/images/%s",
+		ref.Context().RegistryStr(),
+		url.QueryEscape(ref.String()),
+	)
+
+	req, err := http.NewRequest("DELETE", deleteUrl, nil)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Account", accountID)
+	req.Header.Set("Authorization", token)
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
Your request has no request body. You therefore should not need a `Content-Type` header.